### PR TITLE
add clear_on_doubleclick prop for clearing clickData

### PR DESCRIPTION
**Pull request by cpsievert**

### CODE REVIEW

File: src/components/Graph.react.js [hunk 1]
```diff
@@ -212,6 +219,12 @@ PlotlyGraph.propTypes = {
      */
     clickData: PropTypes.object,
 
+    /**
+     * If True, `clear_on_doubleclick` will clear the `clickData` property
+     * when the user double clicks (when dragmode='zoom').
```
Comment (chriddyp) [inline src/components/Graph.react.js]: Only on `dragmode='zoom'`? Also, could we be more specific about `dragmode`, like `figure['layout']['dragmode'] = 'zoom'` or `config['dragmode'] = 'zoom'`

**Comment by chriddyp** [comment]
Seems reasonable and a natural extension of the existing API. I would imagine that `clear_on_click` would be more common / easier to use, why the `doubleclick`?

Ultimately I really want the plotly.js to make this natural behaviour by adding a `click` mode to `select` so that we don't need to program in hacks like this and so that the behaviour is consistent across all contexts (cc @alexcjohnson ), this is being tracked in https://github.com/plotly/plotly.js/issues/1852

**Comment by cpsievert** [comment]
> I would imagine that `clear_on_click` would be more common / easier to use, why the `doubleclick`?

By `clear_on_click`, I assume you mean *any* click event on the plot? It seems as though that would make it too easy to clear a selection, which would be especially problematic if you want selections to persist. In plotly's shiny integration, I've actually been ['clearing click data' on any double-click event (regardless of dragmode)](https://github.com/ropensci/plotly/blob/38f447f6e4e20ae376420305f743f4997d2a9a17/inst/htmlwidgets/plotly.js#L293-L300) for quite some time now and haven't really run into issues with it. Note that d00621f clears `clickData` on `"plotly_deselect"` events as well now which is consistent with that shiny behavior (and should've been included in the initial PR)

**Comment by alexcjohnson** [comment]
@chriddyp @cpsievert can we drop this PR now that plotly.js has built-in click-to-select https://github.com/plotly/plotly.js/pull/2944?

**Comment by chriddyp** [comment]
definitely

**Pull request closed without merging**